lib/Controller/RoomController.php

```diff
@@ -1746,4 +1748,13 @@ public function setMessageExpiration(int $seconds): DataResponse {
 		$this->roomService->setMessageExpiration($this->room, $seconds);
 		return new DataResponse();
 	}
+
+	/**
+	 * @PublicPage
+	 * @RequireModeratorParticipant
+	 */
+	public function setShowHistory(int $showHistory): DataResponse {
+		$this->roomService->setShowHistory($this->room, $showHistory);
```

Comment:
**Required:** 
Since the service can throw after the comment there was resolved, it should be catched here and return a 400 Bad Request